File: src/C#/Creational Patterns/Factory/CreditCards/Extensions/StringExtensions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CreditCards.Extensions
{
    public static class StringExtensions
    {
        public static string SplitPascalCaseString(this string pascalCaseString)
        {
            var modifiedCharacters = pascalCaseString
                .SelectMany((c, i) => i != 0 && char.IsUpper(c) && !char.IsUpper(pascalCaseString[i - 1])
                    ? new char[] { ' ', c } : new char[] { c });

            return new string(modifiedCharacters.ToArray());
        }
    }
}

File: src/Java/Creational Patterns/Factory/CreditCards/src/Extensions/StringExtensions.java
package Extensions;

import java.util.stream.Collectors;
import java.util.stream.Stream;

public class StringExtensions {
    public static String splitPascalCaseString(String pascalCaseString) {
        char[] modifiedCharacters = pascalCaseString
                .chars()
                .mapToObj(c -> (char) c)
                .flatMap(c -> {
                    if (c != ' ' && Character.isUpperCase(c)) {
                        return Stream.of(' ', c);
                    } else {
                        return Stream.of(c);
                    }
                })
                .map(Object::toString)
                .collect(Collectors.joining())
                .toCharArray();

        return new String(modifiedCharacters);
    }
}
